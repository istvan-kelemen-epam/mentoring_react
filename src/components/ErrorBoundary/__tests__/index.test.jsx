//import React from 'react';
import { ErrorBoundary } from '../index.jsx';
import renderer from 'react-test-renderer';

describe('components/ErrorBoundary', () => {
	let errorBoundary;

	beforeEach(() => {
		errorBoundary = new ErrorBoundary();
	});

	describe('.componentDidCatch', () => {
		it('should update state', () => {
			const error = 'error';
			const errorInfo = 'errorInfo';
			errorBoundary.setState = jest.fn();
			errorBoundary.componentDidCatch(error, errorInfo);
			expect(errorBoundary.setState).toHaveBeenCalledWith({
				error: error,
				errorInfo: errorInfo
			});
		});
	});

	describe('.render', () => {
		it('should return children', () => {
			errorBoundary.state.errorInfo = null;
			errorBoundary.props = {
				children: [{}]
			};
			const result = errorBoundary.render();
			expect(result).toEqual(errorBoundary.props.children);
		});

		it('should return error', () => {
			errorBoundary.state.errorInfo = 'errorInfo';
			const result = errorBoundary.render();
			const snapshot = renderer.create(result);
			expect(snapshot).toMatchSnapshot();
		});
	});
});