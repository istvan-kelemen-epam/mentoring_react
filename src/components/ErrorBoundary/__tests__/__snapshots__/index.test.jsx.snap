// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ErrorBoundary .render should return error 1`] = `
<div>
  <h2>
    Something went wrong.
  </h2>
  <details
    style={
      Object {
        "whiteSpace": "pre-wrap",
      }
    }
  >
    <br />
  </details>
</div>
`;
